File: client-flutter/lib/models/profile_data.dart
import 'credit_pack.dart';

class ProfileData {
  final String firstName;
  final String lastName;
  final String? imageUrl;
  final String? email;
  final String? phone;
  final String? birthday;
  final String? gender;
  final List<CreditPack> creditPacks;

  ProfileData({
    required this.firstName,
    required this.lastName,
    this.imageUrl,
    this.email,
    this.phone,
    this.birthday,
    this.gender,
    this.creditPacks = const [],
  });

  String get fullName => '$lastName $firstName';
  String get initials =>
      '${lastName.isNotEmpty ? lastName[0] : ''}${firstName.isNotEmpty ? firstName[0] : ''}'
          .toUpperCase();

  factory ProfileData.fromJson(Map<String, dynamic> json) {
    final rawCredits = json['creditPacks'] ?? json['credit_packs'];
    final creditPacks = <CreditPack>[];
    if (rawCredits is List) {
      for (final c in rawCredits) {
        if (c is Map<String, dynamic>) {
          creditPacks.add(CreditPack.fromJson(c));
        }
      }
    }

    return ProfileData(
      firstName: json['firstname']?.toString() ?? json['firstName']?.toString() ?? '',
      lastName: json['lastname']?.toString() ?? json['lastName']?.toString() ?? '',
      imageUrl: json['photo']?.toString() ?? json['picture']?.toString() ?? json['foto']?.toString() ?? json['imageUrl']?.toString(),
      email: json['email']?.toString(),
      phone: json['phone']?.toString(),
      birthday: json['birthday']?.toString(),
      gender: json['gender']?.toString(),
      creditPacks: creditPacks,
    );
  }
}

File: client-ios/SihlClient/Models/ProfileData.swift
import Foundation

/// Pendant zu `models/profile_data.dart`.
struct ProfileData {
    let firstName: String
    let lastName: String
    let imageUrl: String?
    let email: String?
    let phone: String?
    let birthday: String?
    let gender: String?
    let creditPacks: [CreditPack]

    /// "Nachname Vorname" — wie im Flutter-Original.
    var fullName: String { "\(lastName) \(firstName)".trimmingCharacters(in: .whitespaces) }

    /// Initialen: Anfangsbuchstaben von Nachname + Vorname (Grossbuchstaben).
    var initials: String {
        let l = lastName.first.map(String.init) ?? ""
        let f = firstName.first.map(String.init) ?? ""
        return (l + f).uppercased()
    }

    init(json: [String: Any]) {
        self.firstName = (json["firstname"] as? String) ?? (json["firstName"] as? String) ?? ""
        self.lastName  = (json["lastname"]  as? String) ?? (json["lastName"]  as? String) ?? ""
        self.imageUrl  = (json["photo"] as? String)
                      ?? (json["picture"] as? String)
                      ?? (json["foto"] as? String)
                      ?? (json["imageUrl"] as? String)
        self.email    = json["email"]    as? String
        self.phone    = json["phone"]    as? String
        self.birthday = json["birthday"] as? String
        self.gender   = json["gender"]   as? String

        let raw = json["creditPacks"] ?? json["credit_packs"]
        if let list = raw as? [[String: Any]] {
            self.creditPacks = list.map { CreditPack(json: $0) }
        } else {
            self.creditPacks = []
        }
    }
}
